source/Camera/InfoUtilOther/C++Halcon/HalconConnectUsingZivid/HalconConnectUsingZivid.cpp

```diff
@@ -0,0 +1,103 @@
+#include <Zivid/Zivid.h>
+#include <halconcpp/HalconCpp.h>
+#include <opencv2/core/core.hpp>
+#include <opencv2/highgui/highgui.hpp>
+#include <opencv2/imgproc/imgproc.hpp>
+
+#include <chrono>
+#include <iostream>
+
+
+void savePointCloud(const HalconCpp::HObjectModel3D &model, const std::string filename)
+{
+	const HalconCpp::HString filetype{ "ply" };
+	const HalconCpp::HString filenameCstring{ filename.c_str() };
+	const HalconCpp::HString genParamName{ "invert_normals" };
+	const HalconCpp::HString genParamValue{ "false" };
+	model.WriteObjectModel3d(filetype, filenameCstring, genParamName, genParamValue);
+}
+
+
+HalconCpp::HObjectModel3D toHalconPointCloud(const Zivid::PointCloud &pc)
+{
+	// Extract data from Zivid point cloud
+	const auto width = static_cast<long int>(pc.width());
+	const auto height = static_cast<long int>(pc.height());
+
+	std::vector<float> array1dX(width * height, 0.0);
+	std::vector<float> array1dY(width * height, 0.0);
+	std::vector<float> array1dZ(width * height, 0.0);
+	std::vector<float> array1dR(width * height, 0.0);
+	std::vector<float> array1dG(width * height, 0.0);
+	std::vector<float> array1dB(width * height, 0.0);
+
+	auto data = pc.dataPtr();
+	size_t idx = 0;
+	for (size_t i = 0; i < width; ++i)
+		for (size_t j = 0; j < height; ++j)
+		{
+			idx = width * j + i;
+			if (!isnan(data[idx].x))
+			{
+				array1dX[idx] = data[idx].x;
+				array1dY[idx] = data[idx].y;
+				array1dZ[idx] = data[idx].z;
+				array1dR[idx] = data[idx].red();
+				array1dG[idx] = data[idx].green();
+				array1dB[idx] = data[idx].blue();
+			}
+		}
+
+	// Construct HObjectModel3D from x,y,z data
+	HalconCpp::HImage X{ "real", width, height, array1dX.data() };
+	HalconCpp::HImage Y{ "real", width, height, array1dY.data() };
+	HalconCpp::HImage Z{ "real", width, height, array1dZ.data() };
+	HalconCpp::HObjectModel3D pcHalcon;
+	pcHalcon.XyzToObjectModel3d(X, Y, Z);
+
+	// Color the object based on R,G,B data
+	HalconCpp::HTuple Rtup{ array1dR.data(), width * height };
+	HalconCpp::HTuple Gtup{ array1dG.data(), width * height };
+	HalconCpp::HTuple Btup{ array1dB.data(), width * height };
+	HalconCpp::SetObjectModel3dAttribMod(pcHalcon, "red", "points", Rtup);
+	HalconCpp::SetObjectModel3dAttribMod(pcHalcon, "green", "points", Gtup);
+	HalconCpp::SetObjectModel3dAttribMod(pcHalcon, "blue", "points", Btup);
```

Comment:
No need for variables, use directly.

Reply:
`HalconCpp::SetObjectModel3dAttribMod(pcHalcon, "red", "points", HalconCpp::HTuple{ array1dR.data(), width * height });
	HalconCpp::SetObjectModel3dAttribMod(pcHalcon, "green", "points", HalconCpp::HTuple{ array1dG.data(), width * height });
	HalconCpp::SetObjectModel3dAttribMod(pcHalcon, "blue", "points", HalconCpp::HTuple{ array1dB.data(), width * height });
`